### PR TITLE
Connect api

### PR DESCRIPTION
**Pull request by benwmx**
# Third Milestone 🔢 
in this pull request I added the following changes :

* Added select tag for categories.
* Added categories to the store & a default list.
* Added redux-thunk as middleware to handle API/app interactions.
* Refactored remove book feature to persist your changes in the server.
* Refactored add book feature to persist your changes in the server.

### CODE REVIEW

File: src/redux/books/books.js [hunk 1]
```diff
@@ -1,24 +1,93 @@
...
+const removeBookFromApi = (payload) => (dispatch) => {
+  fetch(`${url}${payload}`, {
+    method: 'DELETE',
+    body: JSON.stringify({ item_id: payload }),
+  })
+    .then((response) => response.text())
+    .then((data) => {
+      if (data === 'The book was deleted successfully!') dispatch(removeBook(payload));
+    });
+};
```
Comment (dasileker) [inline src/redux/books/books.js:47]: - [x] Nice job on the `add & remove` the books from the `API`, but as the requirements say that you need to use `middleware to make async request` not `fetch` method.

mpmpm

**Review by dasileker — CHANGES_REQUESTED** [review]
### **Hi @benwmx,** 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/redux/books/books.js (hunk 1 above)
Comment (benwmx) [reply src/redux/books/books.js:47]: what do you mean exactly?
I'm using the redux-thunk library which is a Middleware that extends the store's abilities and lets you write async logic that interacts with the store.
https://github.com/benwmx/BookStore/blob/398bc58e8d11921b3ca9f01964e7ce97da9e9d5f/src/redux/configureStore.js#L12

and the two functions  Add/Remove, are action creators that return a function ( that can take dispatch & state as arguments )that is executed by redux-thunk, this allows us to do async API calls and any other side effects.
so these two functions are middleware.
https://github.com/benwmx/BookStore/blob/398bc58e8d11921b3ca9f01964e7ce97da9e9d5f/src/redux/books/books.js#L25-L48

**Review by rukundoeric — APPROVED** [review]
Hi @benwmx,

_Excellent job on this project. Keep it up 🎆_

_Congratulations! 🎉  on the splendid job 👍
You can now merge this PR and submit the project completion form 🚀 . I wish you all the very best in your next tasks 🥇_


Cheers and Happy coding!👏👏👏

__Feel free to leave any questions or comments in the PR thread if something is not 100% clear.__

------
>As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

**Comment by benwmx** [comment]
Thank you @rukundoeric for your time reviewing my code.

**Comment by bellom** [comment]
Hello 👋,

This is an additional review as a result of the second opinion request.

I'm sorry this took longer than it should, it totally passed by and got lost in the queue, and also because it was approved already the priority went low.

You're right, you can use redux-thunk.

Keep rocking 🚀!

**Pull request merged**